### PR TITLE
fix(lerobot_local): degrade to passthrough on corrupt model.safetensors

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem
The in-model normalization recovery path in `strands_robots.policies.lerobot_local.processor` calls `_load_checkpoint_state_dict(pretrained_name_or_path)` **unwrapped**, and the helper's own docstring promises it "stays best-effort and never raises into the load path." It only guards the local and Hub `load_file` calls with `except (OSError, ValueError)` / `except (HfHubHTTPError, OSError, ValueError)`.

A truncated or corrupt `model.safetensors` — a common outcome of an interrupted Hugging Face download — makes `safetensors.torch.load_file` raise `safetensors.SafetensorError`, which is a direct `Exception` subclass and is **neither** `OSError` **nor** `ValueError`. It therefore slips past both guards, propagates out of `_load_checkpoint_state_dict`, and crashes the ACT / diffusion policy load instead of degrading to passthrough as designed.

Reproduction (pre-fix):
```python
import tempfile, os
from strands_robots.policies.lerobot_local.processor import _load_checkpoint_state_dict
d = tempfile.mkdtemp()
open(os.path.join(d, "model.safetensors"), "wb").write(b"corrupt")
_load_checkpoint_state_dict(d)   # raises SafetensorError instead of returning None
```

## Change
Add `SafetensorError` to both read-error guards (imported from the public `safetensors` namespace alongside `load_file`, inside the same `ImportError`-guarded block). An unreadable local or Hub checkpoint now returns `None`, letting the caller fall back to the `norm_stats.json` / passthrough path exactly as the surrounding recovery code already documents for the sharded-VLA and offline cases.

## Tests
New `tests/policies/lerobot_local/test_checkpoint_state_dict_load_failsoft.py` pins the fail-soft contract across every branch:
- valid local `model.safetensors` -> returns the state dict (real safetensors round-trip)
- corrupt local `model.safetensors` -> `None` (**fails pre-fix**: `SafetensorError` escaped)
- unreachable Hub (offline / IO error) -> `None`
- valid Hub download -> returns the state dict
- corrupt Hub download -> `None` (**fails pre-fix**)
- safetensors unavailable -> `None`

Green locally: `hatch run lint` clean (ruff + ruff format + mypy, 677 source files) and `MUJOCO_GL=egl` pytest for `tests/policies/lerobot_local/` (496 passed). The two corrupt-checkpoint cases fail on pre-fix code and pass after.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

This PR fixes a real crash bug in the in-model normalization recovery path. `_load_checkpoint_state_dict` in `strands_robots/policies/lerobot_local/processor.py` is called unwrapped and is documented as best-effort, but its two `load_file` guards only caught `(OSError, ValueError)` (local) and `(HfHubHTTPError, OSError, ValueError)` (Hub). A truncated/corrupt `model.safetensors` raises `safetensors.SafetensorError`, which is a direct `Exception` subclass and neither `OSError` nor `ValueError`, so it escaped both guards and crashed the ACT/diffusion load instead of degrading to passthrough. The fix imports `SafetensorError` from the public `safetensors` namespace inside the existing `ImportError`-guarded block and adds it to both `except` tuples.

I verified the core claims independently: `SafetensorError` is importable from `safetensors` and `issubclass(SafetensorError, (OSError, ValueError))` is `False`, and `load_file` on corrupt bytes does raise `SafetensorError`, confirming the pre-fix leak. The import placement is correct — it lives in the same `try/except ImportError` block as `load_file`, so an environment without `safetensors` returns `None` before the widened `except` clauses are ever evaluated (no `NameError` risk). The change is narrow, adds no new user-facing surface, and keeps exception clauses explicit rather than reaching for `except Exception`, consistent with AGENTS.md's "Exception Clauses Must Be Narrow."

## What's good

- Minimal, surgical diff (+8/-3) that fixes the exact reported failure without scope creep.
- Explicit narrow exception widening (adds `SafetensorError` by name, not a broad catch-all).
- New tests pin the fail-soft contract across all branches (valid/corrupt local, unreachable/valid/corrupt Hub, safetensors-unavailable), with the two corrupt cases documented as failing pre-fix.
- No emojis, no host paths, private helper stays private.
